eng/scripts/New-ChangelogEntry.ps1

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env pwsh
```

Comment:
Before we go an implement all this ourselves, have we looked to see if there exist any tools that will do this sort of thing for us already?

If we do implement we should take a look to see if we can use anything from https://github.com/microsoft/mcp/blob/main/eng/common/scripts/ChangeLog-Operations.ps1 to share code.